client/src/components/NodesEntriesSelector/NodesEntriesSelector.tsx

```diff
@@ -0,0 +1,132 @@
+import ActionButton, {ActionButtonColor} from 'components/ActionButton/ActionButton.tsx';
+
+interface NodesEntriesSelectorProps {
+    selectedNodes: NodeEntry[];
+    setSelectedNodes: React.Dispatch<React.SetStateAction<NodeEntry[]>>;
+    nodesToExclude: NodeEntry[];
+    onAdd: () => void;
+    onClose: () => void;
+}
+
+const NodesEntriesSelector: React.FC<NodesEntriesSelectorProps> = ({
+                                                                       selectedNodes,
+                                                                       setSelectedNodes,
+                                                                       nodesToExclude,
+                                                                       onAdd,
+                                                                       onClose,
+                                                                   }) => {
+    const [nodes, setNodes] = useState<NodeEntry[]>([]);
+    const [selectAll, setSelectAll] = useState<boolean>(false);
+
+    useEffect(() => {
+        const fetchNodes = async () => {
+            try {
+                const data = await ManagmentServiceApiInstance.getNodes();
+                const rows = data.map((node) => ({
+                    name: node.name,
+                    running: node.running,
+                    accuracy: node.accuracy,
+                    customPrompt: node.customPrompt,
+                    updated: node.updated,
+                    added: node.added,
+                }));
+                setNodes(rows);
+            } catch (error) {
+                console.error('Failed to fetch nodes:', error);
+            }
+        };
+        fetchNodes();
+    }, []);
+
+    const availableNodes = nodes.filter(
+        (node) => !nodesToExclude.some((excluded) => excluded.name === node.name)
+    );
+
+    useEffect(() => {
+        setSelectAll(
+            availableNodes.length > 0 && selectedNodes.length === availableNodes.length
+        );
+    }, [selectedNodes, availableNodes]);
+
+    const handleSelectAllChange = () => {
+        if (selectAll) {
+            setSelectedNodes([]);
+        } else {
+            setSelectedNodes(availableNodes);
+        }
+        setSelectAll(!selectAll);
+    };
+
+    const handleCheckboxChange = (node: NodeEntry) => {
+        setSelectedNodes((prevSelected) => {
+            const isSelected = prevSelected.some(
+                (selectedNode) => selectedNode.name === node.name
+            );
+            return isSelected
+                ? prevSelected.filter((selectedNode) => selectedNode.name !== node.name)
+                : [...prevSelected, node];
+        });
+    };
+
+    const columns: TableColumn<NodeEntry>[] = [
+        {
+            header: (
+                <Checkbox
+                    checked={selectAll}
+                    onChange={handleSelectAllChange}
+                />
+            ),
```

Comment:
I thought about adding support for checkboxes into the table itself, but if this works well we can keep it